core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -370,9 +381,27 @@ class LogManager(val logDirs: Array[File],
   }
 
   /**
-   *  Delete a log.
+   *  Delete logs marked for deletion.
    */
-  def deleteLog(topicAndPartition: TopicAndPartition) {
+  private def deleteLogs(): Unit = {
```

Comment:
Can you clarify how this interacts with `cleanupSegments`? I haven’t investigated thoroughly but it appears `deleteSegments` can run on the moved directory. So that segment deletion can run asynchronously in the scheduler which can fail (if the directory deletion happens earlier)